src/gordon_gcp/plugins/event_consumer.py

```diff
@@ -30,15 +30,243 @@
+        self.error_channel = error_channel
+        self.kwargs = kwargs
+
+    def _validate_config(self):
+        errors = []
+        # req keys: keyfile, project, topic, subscription
+        # TODO (lynn): keyfile won't be required once we support other
+        #              auth methods
+        if not self.config.get('keyfile'):
+            msg = ('The path to a Service Account JSON keyfile is required to '
+                   'authenticate for Google Cloud Pub/Sub.')
+            errors.append(msg)
+
+        if not self.config.get('project'):
+            msg = ('The GCP project where Cloud Pub/Sub is located is '
+                   'required.')
+            errors.append(msg)
+
+        if not self.config.get('topic'):
+            msg = ('A topic for the client to subscribe to in Cloud Pub/Sub is '
+                   'required.')
+            errors.append(msg)
+
+        if not self.config.get('subscription'):
+            msg = ('A subscription for the client to pull messages from in '
+                   'Cloud Pub/Sub is required.')
+            errors.append(msg)
+
+        if errors:
+            exp_msg = 'Invalid configuration:\n'
+            for error in errors:
+                logging.error(error)
+                exp_msg += error + '\n'
+            raise exceptions.GCPConfigError(exp_msg)
+
+        topic_prefix = f'projects/{self.config["project"]}/topics/'
+        if not self.config.get('topic').startswith(topic_prefix):
+            self.config['topic'] = f'{topic_prefix}{self.config["topic"]}'
+
+        sub_prefix = f'projects/{self.config["project"]}/subscriptions/'
+        if not self.config.get('subscription').startswith(sub_prefix):
+            sub = f'{sub_prefix}{self.config["subscription"]}'
+            self.config['subscription'] = sub
+
+    def _init_auth(self):
+        # a publisher client can't be made with credentials if a channel is
+        # already made/provided, which is what happens when the emulator is
+        # running ಠ_ಠ
+        # See (https://github.com/GodsaogleCloudPlatform/
+        #      google-cloud-python/pull/4839)
+        auth_client = None
+        if not os.environ.get('PUBSUB_EMULATOR_HOST'):
+            scopes = self.config.get('scopes')
+            # creating a dummy `session` as the pubsub.PublisherClient never
+            # uses it but without it aiohttp will complain about an unclosed
+            # client session that would otherwise be made by default
+            auth_client = auth.GAuthClient(
+                keyfile=self.config['keyfile'], scopes=scopes, session='noop')
+        return auth_client
+
+    def _init_subscriber_client(self, auth_client):
+        # Silly emulator constraints
+        creds = getattr(auth_client, 'creds', None)
+        client = pubsub.SubscriberClient(credentials=creds)
+
+        try:
+            client.create_subscription(
+                self.config['subscription'], self.config['topic'])
```

Comment:
Does Google log anything when this happens?  Otherwise it could be good to log a message like "Starting a subcriber for {subscription} on {topic}" or similar, so someone can easily debug if they accidentally have a typo or write the wrong thing in their config,  so everything looks like it's working, but they're still not getting the messages they expect.

Reply:
Good idea, yeah the google client does not.